### DISCUSSION TITLE
Using class-transformer with polytype

### DISCUSSION THREAD

**Discussion by aru-py** [fasttime/Polytype#13] [Q&A] [answered]
I am trying to convert an object literal into a class extended with polytype using class-transformer, but I'm running into the following issue:

```
TypeError: Cannot read property 'prototype' of null
```

The below code reproduces this error.

````
import { classes } from "polytype";
import {plainToInstance} from "class-transformer";

class A {}
class B {}

// target class
class C extends classes(A,B) {}

// throws error
const obj = plainToInstance(C, {})

**Comment by fasttime** [comment, marked as answer]
I believe that class-transformer is assuming that every constructor has a prototype. I don't know why they are doing that, but I can think of other cases where that code would fail. You may want to report this issue to class-transformer.

```js
import { plainToInstance } from "class-transformer";

function Base() { }
Object.setPrototypeOf(Base, null);
class Derived extends Base { }

// throws error
const obj = plainToInstance(Derived, { });
```

Note that in this example, `Derived` is a valid, instantiable and subclassable class.

**Reply by aru-py** [reply]
Thank you – I'll raise an issue there then!
